### ISSUE TITLE
An invalid response was received from the upstream server.

### ISSUE THREAD

**Issue by nor-bu** [Mashape/kong#2005]
I have setup a dev kong server and api server.

the kong server run under alphaapi.com and the api server api.alphaapi.com.
When i add a api alphaapi.com/test -> api.alphaapi.com/app1 i got this error:
"An invalid response was received from the upstream server."

api.alphaapi.com is running a nginx proxy for local node.js apps.

when i call the api.alphaapi.com then it works all fine.

What is the problem?

**Comment by pouledodue** [comment]
do you have a message in kong logs

**Comment by Tieske** [comment]
please post the exact config and commands you used. Also; please respect the template, as we're now asking for information that was already requested in the issue template.

**Comment by nor-bu** [comment]
Hi sorry for the late answer, this is in the log:
```
==> /usr/local/kong/logs/error.log <==
2017/01/22 16:53:26 [notice] 10863#0: signal 17 (SIGCHLD) received

==> /usr/local/kong/logs/access.log <==
127.0.0.1 - - [22/Jan/2017:16:53:26 +0000] "POST /cluster/events/ HTTP/1.1" 200 5 "-" "lua-resty-http/0.08 (Lua) ngx_lua/10006"
```

This is my config file.

```
# -----------------------
# Kong configuration file
# -----------------------
#
# The commented-out settings shown in this file represent the default values.
#
# This file is read when `kong start` or `kong compile` are used. Kong
# generates the Nginx configuration with the settings specified in this file.
#
# All environment variables prefixed with `KONG_` and capitalized will override
# the settings specified in this file.
# Example:
#   `log_level` setting -> `KONG_LOG_LEVEL` env variable
#
# Boolean values can be specified as `on`/`off` or `true`/`false`.
# Lists must be specified as comma-separated strings.
#
# All comments in this file can be removed safely, including the
# commented-out properties.
# You can verify the integrity of your settings with `kong check <conf>`.

#------------------------------------------------------------------------------
# GENERAL
#------------------------------------------------------------------------------

#prefix = /usr/local/kong/       # Working directory. Equivalent to Nginx's
                                 # prefix path, containing temporary files
                                 # and logs.
                                 # Each Kong process must have a separate
                                 # working directory.

#log_level = notice              # Log level of the Nginx server. Logs are
                                 # found at <prefix>/logs/error.log
# Note: See http://nginx.org/en/docs/ngx_core_module.html#error_log for a list
# of accepted values.

#custom_plugins =                # Comma-separated list of additional plugins
                                 # this node should load.
                                 # Use this property to load custom plugins
                                 # that are not bundled with Kong.
                                 # Plugins will be loaded from the
                                 # `kong.plugins.{name}.*` namespace.

#anonymous_reports = on          # Send anonymous usage data such as error
                                 # stack traces to help improve Kong.

#------------------------------------------------------------------------------
# NGINX
#------------------------------------------------------------------------------

#proxy_listen = 0.0.0.0:80     # Address and port on which Kong will accept
                                 # HTTP requests.
                                 # This is the public-facing entrypoint of
                                 # Kong, to which your consumers will make
                                 # requests to.
# Note: See http://nginx.org/en/docs/http/ngx_http_core_module.html#listen for
# a description of the accepted formats for this and other *_listen values.

#proxy_listen_ssl = 0.0.0.0:8443 # Address and port on which Kong will accept
                                 # HTTPS requests if `ssl` is enabled.

#admin_listen = 0.0.0.0:8001     # Address and port on which Kong will expose
                                 # an entrypoint to the Admin API.
                                 # This API lets you configure and manage Kong,
                                 # and should be kept private and secured.

#nginx_worker_processes = auto   # Determines the number of worker processes
                                 # spawned by Nginx.

#nginx_daemon = on               # Determines wether Nginx will run as a daemon
                                 # or as a foreground process. Mainly useful
                                 # for development or when running Kong inside
                                 # a Docker environment.

#mem_cache_size = 128m           # Size of the in-memory cache for database
                                 # entities. The accepted units are `k` and
                                 # `m`, with a minimum recommended value of
                                 # a few MBs.

#ssl = on                        # Determines if Nginx should be listening for
                                 # HTTPS traffic on the `proxy_listen_ssl`
                                 # address. If disabled, Nginx will only bind
                                 # itself on `proxy_listen`, and all SSL
                                 # settings will be ignored.

#ssl_cert =                      # If `ssl` is enabled, the absolute path to
                                 # the SSL certificate for the
                                 # `proxy_listen_ssl` address.

#ssl_cert_key =                  # If `ssl` is enabled, the absolute path to
                                 # the SSL key for the `proxy_listen_ssl`
                                 # address.

#------------------------------------------------------------------------------
# DATASTORE
#------------------------------------------------------------------------------

# Kong will store all of its data (such as APIs, consumers and plugins) in
# either Cassandra or PostgreSQL.
#
# All Kong nodes belonging to the same cluster must connect themselves to the
# same database.

#database = postgres             # Determines which of PostgreSQL or Cassandra
                                 # this node will use as its datastore.
                                 # Accepted values are `postgres` and
                                 # `cassandra`.

#pg_host = 127.0.0.1             # The PostgreSQL host to connect to.
#pg_port = 5432                  # The port to connect to.
#pg_user = kong                  # The username to authenticate if required.
pg_password = k0ng              # The password to authenticate if required.
#pg_database = kong              # The database name to connect to.

#pg_ssl = off                    # Toggles client-server TLS connections
                                 # between Kong and PostgreSQL.

#pg_ssl_verify = off             # Toggles server certificate verification if
                                 # `pg_ssl` is enabled.
                                 # See the `lua_ssl_trusted_certificate`
                                 # setting to specify a certificate authority.

#cassandra_contact_points = 127.0.0.1  # A comma-separated list of contact
                                       # points to your cluster.

#cassandra_port = 9042           # The port on which your nodes are listening
                                 # on. All your nodes and contact points must
                                 # listen on the same port.

#cassandra_keyspace = kong       # The keyspace to use in your cluster.

#cassandra_consistency = ONE     # Consistency setting to use when reading/
                                 # writing to the Cassandra cluster.

#cassandra_timeout = 5000        # Defines the timeout (in ms), for reading
                                 # and writing.

#cassandra_ssl = off             # Toggles client-to-node TLS connections
                                 # between Kong and Cassandra.

#cassandra_ssl_verify = off      # Toggles server certificate verification if
                                 # `cassandra_ssl` is enabled.
                                 # See the `lua_ssl_trusted_certificate`
                                 # setting to specify a certificate authority.

#cassandra_username = kong       # Username when using the
                                 # `PasswordAuthenticator` scheme.

#cassandra_password = kong       # Password when using the
                                 # `PasswordAuthenticator` scheme.

#cassandra_repl_strategy = SimpleStrategy  # When migrating for the first time,
                                           # Kong will use this setting to
                                           # create your keyspace.
                                           # Accepted values are
                                           # `SimpleStrategy` and
                                           # `NetworkTopologyStrategy`.

#cassandra_repl_factor = 1       # When migrating for the first time, Kong
                                 # will create the keyspace with this
                                 # replication factor when using the
                                 # `SimpleStrategy`.

#cassandra_data_centers = dc1:2,dc2:3  # When migrating for the first time,
                                       # will use this setting when using the
                                       # `NetworkTopologyStrategy`.
                                       # The format is a comma-separated list
                                       # made of <dc_name>:<repl_factor>.

#------------------------------------------------------------------------------
# CLUSTERING
#------------------------------------------------------------------------------

# In addition to pointing to the same database, each Kong node must join the
# same cluster.
#
# Kong's clustering works on the IP layer (hostnames are not supported, only
# IPs) and expects a flat network topology without any NAT between the
# datacenters.
#
# A common pattern is to create a VPN between the two datacenters such that
# the flat network assumption is not violated.
#
# See the clustering reference for more informations:
# https://getkong.org/docs/latest/clustering/

#cluster_listen = 0.0.0.0:7946   # Address and port used to communicate with
                                 # other nodes in the cluster.
                                 # All other Kong nodes in the same cluster
                                 # must be able to communicate over both
                                 # TCP and UDP on this address.
                                 # Only IPv4 addresses are supported.

#cluster_listen_rpc = 127.0.0.1:7373  # Address and port used by this node to
                                      # communicate with the cluster.
                                      # Only contains TCP traffic over the
                                      # local network.

#cluster_advertise =             # By default, the `cluster_listen` address
                                 # is advertised over the cluster.
                                 # If the `cluster_listen` host is '0.0.0.0',
                                 # then the first local, non-loopback IPv4
                                 # address will be advertised to other nodes.
                                 # However, in some cases (specifically NAT
                                 # traversal), there may be a routable address
                                 # that cannot be bound to. This flag enables
                                 # advertising a different address to support
                                 # this.

#cluster_encrypt_key =           # base64-encoded 16-bytes key to encrypt
                                 # cluster traffic with.

#cluster_ttl_on_failure = 3600   # Time to live (in seconds) of a node in the
                                 # cluster when it stops sending healthcheck
                                 # pings, possibly caused by a node or network
                                 # failure.
                                 # If a node is not able to send a new
                                 # healthcheck ping before the expiration,
                                 # other nodes in the cluster will stop
                                 # attempting to connect to it.
                                 # Recommended to be at least `60`.

#cluster_profile = wan           # The timing profile for inter-cluster pings
                                 # and timeouts. If a `lan` or `local` profile
                                 # is used over the Internet, a high rate of
                                 # failures is risked as the timing contraints
                                 # would be too tight.
                                 # Accepted values are `local`, `lan`, `wan`.

#------------------------------------------------------------------------------
# DNS RESOLVER
#------------------------------------------------------------------------------

#dnsmasq = off                    # Toggles if Kong should start/stop dnsmasq,
                                 # which can be used as the Nginx DNS resolver.
                                 # Using dnsmasq allows Nginx to resolve
                                 # domains defined in /etc/hosts.
                                 # dnsmasq must be installed and available in
                                 # your $PATH.

#dnsmasq_port = 8053             # The port on which dnsmasq should listen to
                                 # for queries.

#dns_resolver = 8.8.8.8          # Configure a name server to be used by Nginx.
                                 # Only valid when `dnsmasq` is disabled.

#------------------------------------------------------------------------------
# DEVELOPMENT & MISCELLANEOUS
#------------------------------------------------------------------------------

# Additional settings inherited from lua-nginx-module allowing for more
# flexibility and advanced usage.
#
# See the lua-nginx-module documentation for more informations:
# https://github.com/openresty/lua-nginx-module

#lua_ssl_trusted_certificate =   # Absolute path to the certificate
                                 # authority file for Lua cosockets in PEM
                                 # format. This certificate will be the one
                                 # used for verifying Kong's database
                                 # connections, when `pg_ssl_verify` or
                                 # `cassandra_ssl_verify` are enabled.

#lua_ssl_verify_depth = 1        # Sets the verification depth in the server
                                 # certificates chain used by Lua cosockets,
                                 # set by `lua_ssl_trusted_certificate`.
                                 # This includes the certificates configured
                                 # for Kong's database connections.

#lua_code_cache = on             # When disabled, every request will run in a
                                 # separate Lua VM instance: all Lua modules
                                 # will be loaded from scratch. Useful for
                                 # adopting an edit-and-refresh approach while
                                 # developing a plugin.
                                 # Turning this directive off has a severe
                                 # impact on performance.

#lua_package_path =              # Sets the Lua module search path (LUA_PATH).
                                 # Useful when developing or using custom
                                 # plugins not stored in the default search
                                 # path.

#lua_package_cpath =             # Sets the Lua C module search path
                                 # (LUA_CPATH).
```

This is the response from alphaapi.com:8001/apis

```
{
  "data": [
    {
      "upstream_url": "http://alphaasateapis.com/test",
      "created_at": 1485021077000,
      "request_path": "/test",
      "id": "59bfd026-5960-463d-9fde-bdeadb624f68",
      "name": "Test-Api",
      "preserve_host": false,
      "strip_request_path": false,
      "request_host": "test.alphaasateapis.com"
    }
  ],
  "total": 1
}
```

**Comment by nor-bu** [comment]
so now i have time, i have setup kong completely fresh, now i get a clear error.

when i send a GET request with postman or the browser to alphaapis.com/hello i saw this in the log this and get this error: 400 Bad Request - Request Header Or Cookie Too Large


```
00.000.000.00 - - [25/Jan/2017:06:03:25 +0000] "GET /hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello/hello HTTP/1.0" 400 255 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/55.0.2883.95 Safari/537.36"
```

**Comment by Tieske** [comment]
```
==> /usr/local/kong/logs/error.log <==
2017/01/22 16:53:26 [notice] 10863#0: signal 17 (SIGCHLD) received

==> /usr/local/kong/logs/access.log <==
127.0.0.1 - - [22/Jan/2017:16:53:26 +0000] "POST /cluster/events/ HTTP/1.1" 200 5 "-" "lua-resty-http/0.08 (Lua) ngx_lua/10006"
```
This is normal.

```
Request Header Or Cookie Too Large
```
That doesn't seem normal. What version of Kong are you using? And can you post the relevant part of the logs?

**Comment by nor-bu** [comment]
The "400 Bad request - Request Header Or Cookie to large" is the response from the /hello request
```
<html>
    <head>
        <title>400 Request Header Or Cookie Too Large</title>
    </head>
    <body bgcolor="white">
        <center>
            <h1>400 Bad Request</h1>
        </center>
        <center>Request Header Or Cookie Too Large</center>
        <hr>
        <center>openresty/1.11.2.1</center>
    </body>
</html>
```

In the logs are so much entries like this:
```

00.00.00.00 - - [28/Jan/2017:06:42:17 +0000] "GET /hello/hello/hello/hello HTTP/1.0" 400 255 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/55.0.2883.95 Safari/537.36"
00.00.00.00 - - [28/Jan/2017:06:42:17 +0000] "GET /hello/hello/hello HTTP/1.0" 400 255 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/55.0.2883.95 Safari/537.36"
00.00.00.00 - - [28/Jan/2017:06:42:17 +0000] "GET /hello/hello HTTP/1.0" 400 255 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/55.0.2883.95 Safari/537.36"
80.75.112.22 - - [28/Jan/2017:06:42:17 +0000] "GET /hello HTTP/1.1" 400 255 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/55.0.2883.95 Safari/537.36"
```

you see there are more and more /hello this is from one request. In the logs start with i thin 100 /hello/hello in one log entry like this from my last comment and go down this last three. There is no other logs.

My kong version is 0.9.8 on ubuntu 16.04 installed over package (xenial_all). no custom code

**Comment by nor-bu** [comment]
Any idea?

**Comment by Tieske** [comment]
looks like Kong is proxying the request to itself.

if you change the config with `strip_request_path=true` it will no longer append the extra `/hello` probably. Easy to try.

My guess is that `test.alphaasateapis.com` points back to Kong. Is your backend service running on the same server on a different port you forgot to include in the upstream url?

**Comment by nor-bu** [comment]
This is the api config:

```
{
  "data": [
    {
      "upstream_url": "http://alphaapis.com",
      "created_at": 1485324115000,
      "request_path": "/",
      "id": "dade3922-4f57-405c-bf16-eb5ab185f76a",
      "name": "test-api",
      "preserve_host": false,
      "strip_request_path": true,
      "request_host": "test.alphaapis.com"
    }
  ],
  "total": 1
}
```

This is now the log:
```
00.00.00.00- - [01/Feb/2017:09:24:46 +0000] "GET /hello HTTP/1.0" 400 255 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/55.0.2883.95 Safari/537.36"
00.00.00.00 - - [01/Feb/2017:09:24:46 +0000] "GET /hello HTTP/1.0" 400 255 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/55.0.2883.95 Safari/537.36"
00.00.00.00 - - [01/Feb/2017:09:24:46 +0000] "GET /hello HTTP/1.0" 400 255 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/55.0.2883.95 Safari/537.36"
00.00.00.00 - - [01/Feb/2017:09:24:46 +0000] "GET /hello HTTP/1.0" 400 255 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/55.0.2883.95 Safari/537.36"
00.00.00.00 - - [01/Feb/2017:09:24:46 +0000] "GET /hello HTTP/1.1" 400 255 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/55.0.2883.95 Safari/537.36"
```

The test.alphaapis.com is separate server with a node.js app that response the request path, when i call the api from the kong server with curl it works fine.

**Comment by Tieske** [comment]
you have something in your setup that makes the request go in circles through Kong. The new log proves it as changing the 'stripping' now results in only the simple path. 

Reconfigure based on IP addresses instead of names and see if that works.

**Comment by nor-bu** [comment]
It not working, the logs:

```
80.75.112.22 - - [01/Feb/2017:09:42:27 +0000] "GET /hello HTTP/1.0" 400 255 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/55.0.2883.95 Safari/537.36"
80.75.112.22 - - [01/Feb/2017:09:42:27 +0000] "GET /hello HTTP/1.0" 400 255 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/55.0.2883.95 Safari/537.36"
80.75.112.22 - - [01/Feb/2017:09:42:27 +0000] "GET /hello HTTP/1.1" 400 255 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_2) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/55.0.2883.95 Safari/537.36"
```

The response:

```
<html>
    <head>
        <title>400 Request Header Or Cookie Too Large</title>
    </head>
    <body bgcolor="white">
        <center>
            <h1>400 Bad Request</h1>
        </center>
        <center>Request Header Or Cookie Too Large</center>
        <hr>
        <center>openresty/1.11.2.1</center>
    </body>
</html>
```

The Api config:

```
{
  "data": [
    {
      "upstream_url": "http://000.000.000.000",
      "created_at": 1485324115000,
      "request_path": "/",
      "id": "dade3922-4f57-405c-bf16-eb5ab185f76a",
      "name": "test-api",
      "preserve_host": false,
      "strip_request_path": true,
      "request_host": "000.000.000.000"
    }
  ],
  "total": 1
}
```

When i send curl request to the request IP from kong server it works. The api server is now a nodejs app listen on port 80

**Comment by nor-bu** [comment]
I have now install kong on 14.04 instead of 16.04, it dosn't work but i get a other error:

```
2017/02/01 21:59:09 [crit] 6270#0: accept4() failed (24: Too many open files)
2017/02/01 21:59:10 [crit] 6270#0: accept4() failed (24: Too many open files)
2017/02/01 21:59:10 [crit] 6270#0: accept4() failed (24: Too many open files)
2017/02/01 21:59:11 [crit] 6270#0: accept4() failed (24: Too many open files)
2017/02/01 21:59:11 [error] 6270#0: *7226 lua entry thread aborted: runtime error: /usr/local/share/lua/5.1/pl/utils.lua:232: unable to generate a unique filename
stack traceback:
coroutine 0:
	[C]: in function 'tmpname'
	/usr/local/share/lua/5.1/pl/utils.lua:232: in function 'executeex'
	/usr/local/share/lua/5.1/kong/serf.lua:38: in function 'event'
	/usr/local/share/lua/5.1/kong/core/hooks.lua:154: in function 'fn'
	/usr/local/share/lua/5.1/kong/core/events.lua:25: in function 'fn'
	/usr/local/share/lua/5.1/mediator.lua:103: in function 'publish'
	/usr/local/share/lua/5.1/kong/core/events.lua:33: in function 'publish'
	/usr/local/share/lua/5.1/kong/dao/dao.lua:76: in function 'event'
	/usr/local/share/lua/5.1/kong/dao/dao.lua:294: in function 'update'
	/usr/local/share/lua/5.1/kong/core/cluster.lua:129: in function </usr/local/share/lua/5.1/kong/core/cluster.lua:106>, context: ngx.timer
2017/02/01 21:59:11 [crit] 6270#0: accept4() failed (24: Too many open files)
```

Is this more help full?

**Comment by Tieske** [comment]
what curl command did you use?

**Comment by nor-bu** [comment]
Simple get request

`curl http://test.alphaapis.com`

**Comment by machbio** [comment]
I have a similar error - I am running the Kong behind a Corporate Firewall - here is my Docker-compose file;

```
kong-database:
  image: postgres:9.4
  ports:
    - 5432
  environment:
    - POSTGRES_USER=kong
    - POSTGRES_DB=kong

kong:
  image: kong:latest
  restart: always
  links:
    - kong-database:kong-database
  ports:
    - 8000:8000
    - 8443:8443
    - 8001:8001
  environment:
    - KONG_DATABASE=postgres
    - KONG_PG_HOST=kong-database
```

The Error Log ;

> ==> /usr/local/kong/logs/error.log <==
2017/02/07 19:34:40 [error] 81#0: *195 mockbin.com could not be resolved (3: Host not found), client: 172.17.0.1, server: kong, request: "GET / HTTP/1.1", host: "mockbin.com"

But when I run the Curl on the Mockbin API directly, I get the expected result, but when I go through Kong - I get the following error similar to this ticket

> An invalid response was received from the upstream server

**Comment by Tieske** [comment]
@nor-bu sorry for the late reply. This bug was a side effect of the upstream hostname being overwritten by the incoming hostname, causing the circular references.

Has been fixed in new RC, please try. Closing this now, please reopen if the problem reoccurs.
